nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PackageFlowFile.java

```diff
@@ -0,0 +1,146 @@
+                    + " Multiple input queues can produce unexpected batching behavior.")
+            .required(true)
+            .defaultValue("1")
+            .addValidator(StandardValidators.createLongValidator(1, 10_000, true))
+            .build();
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("The packaged FlowFile is sent to this relationship")
+            .build();
+    static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("The FlowFiles that were used to create the package are sent to this relationship")
+            .build();
+
+    private static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(
+            new LinkedHashSet<>(Arrays.asList(
+                    REL_SUCCESS,
+                    REL_ORIGINAL
+            )));
+
+    private static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = Collections.unmodifiableList(
+            Arrays.asList(
+                    BATCH_SIZE
+            ));
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTY_DESCRIPTORS;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        final List<FlowFile> flowFiles = session.get(context.getProperty(BATCH_SIZE).asInteger());
+        if (flowFiles.isEmpty()) {
+            return;
+        }
+
+        final Map<String, String> attributes = new HashMap<>();
+        attributes.put(CoreAttributes.MIME_TYPE.key(), StandardFlowFileMediaType.VERSION_3.getMediaType());
+        attributes.put(CoreAttributes.FILENAME.key(), createFilename(flowFiles));
+
+        final FlowFilePackager packager = new FlowFilePackagerV3();
+
+        FlowFile packagedFlowFile = session.create(flowFiles);
+        packagedFlowFile = session.write(packagedFlowFile, out -> {
+            try (final OutputStream bufferedOut = new BufferedOutputStream(out)) {
+                for (final FlowFile flowFile : flowFiles) {
+                    session.read(flowFile, in -> {
+                        try (final InputStream bufferedIn = new BufferedInputStream(in)) {
+                            packager.packageFlowFile(bufferedIn, bufferedOut, flowFile.getAttributes(), flowFile.getSize());
+                        }
+                    });
+                }
+            }
+        });
+
+        packagedFlowFile = session.putAllAttributes(packagedFlowFile, attributes);
+        session.transfer(packagedFlowFile, REL_SUCCESS);
+        session.transfer(flowFiles, REL_ORIGINAL);
+    }
+
+    private String createFilename(final List<FlowFile> flowFiles) {
```

Comment:
Do you think users would prefer more positive control over the name of the output filename, perhaps via an attribute?

Reply:
For keeping the functionality simple, it may be better to leave the filename unchanged. A subsequent UpdateAttribute Processor could handle changes if needed.